app/auth/views/proton.py

```diff
@@ -50,12 +50,8 @@ def proton_login():
         return redirect(url_for("auth.login"))
 
     next_url = sanitize_next_url(request.args.get("next"))
-    if next_url:
-        redirect_uri = _redirect_uri + "?next=" + encode_url(next_url)
-    else:
-        redirect_uri = _redirect_uri
-
-    proton = OAuth2Session(PROTON_CLIENT_ID, redirect_uri=redirect_uri)
+    session["oauth_next"] = next_url
```

Comment:
What if there's no `next`?

Reply:
Then a `None` will be stored

Reply:
If the session object expects a `str` and `None` is stored then `"None"` will be stored won't it? and that's not a valid url

Reply:
Stored only if `next_url` is not None in order to make it more explicit